framework/src/common/errors.rs

```diff
@@ -82,64 +81,34 @@ pub enum NetBricksError {
 #[macro_export]
 macro_rules! error_chain {
     ($error:expr) => {
-        $crate::common::errors::print_chain($error, log::LevelFilter::Error, true)
+        error!("{}", $crate::common::errors::string_chain($error))
     };
-    () => {};
-}
-
-#[macro_export]
-macro_rules! error_causes {
-    ($error:expr) => {
-        $crate::common::errors::print_chain($error, log::LevelFilter::Error, false)
-    };
-    () => {};
 }
 
 #[macro_export]
 macro_rules! warn_chain {
     ($error:expr) => {
-        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, true)
-    };
-    () => {};
-}
-
-#[macro_export]
-macro_rules! warn_causes {
-    ($error:expr) => {
-        $crate::common::errors::print_chain($error, log::LevelFilter::Warn, false)
+        warn!("{}", $crate::common::errors::string_chain($error))
     };
-    () => {};
 }
 
 /// Read a `failure` `Error` and print out the causes and a backtrace as
 /// `log::error`s
-#[doc(hidden)]
-pub fn print_chain(e: &Error, log_level: log::LevelFilter, with_backtrace: bool) {
-    if log_level == log::LevelFilter::Warn {
-        warn!("{}", e);
-        for cause in e.iter_causes() {
-            warn!("caused by: {}", cause);
-        }
-    } else if log_level == log::LevelFilter::Error {
-        error!("{}", e);
-        for cause in e.iter_causes() {
-            error!("caused by: {}", cause);
+pub fn string_chain(e: &Error) -> String {
+    let mut error = e.to_string();
+    for (i, cause) in e.iter_causes().enumerate() {
+        if i == 0 {
+            error.push_str("\n")
         }
-    } else {
-        return ();
+        error.push_str(&format!("Cause: {}\n", cause));
     }
 
-    match (
-        ::std::env::var("RUST_BACKTRACE")
-            .as_ref()
-            .map(|s| s.as_str()),
-        with_backtrace,
-        log_level,
-    ) {
-        (Ok("1"), _, log::LevelFilter::Error) => error!("Backtrace:\n{}", e.backtrace()),
-        (Ok("1"), _, log::LevelFilter::Warn) => warn!("Backtrace:\n{}", e.backtrace()),
-        (Ok("0"), true, log::LevelFilter::Error) => error!("Backtrace:\n{}", e.backtrace()),
-        (Ok("0"), true, log::LevelFilter::Warn) => warn!("Backtrace:\n{}", e.backtrace()),
-        _ => (),
+    if let Ok("1") = ::std::env::var("RUST_BACKTRACE")
+        .as_ref()
+        .map(|s| s.as_str())
+    {
+        error.push_str(&format!("Backtrace:\n{}", e.backtrace()))
```

Comment:
There could be some line breaks quirkiness here I think if there are no `causes`.